fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -645,4 +737,19 @@ private void checkLinkForLdpResourceCreation(final String link) {
         }
     }
 
+    /**
+     * Parse the RFC-3230 Digest response header value.  Look for a
+     * sha1 checksum and return it as a urn, if missing or malformed
+     * an empty string is returned.
+     * @param digest The Digest header value
+     * @return the sha1 checksum value
+     */
+    private String parseDigestHeader(final String digest) {
+        final Map<String,String> digestPairs = RFC3230_SPLITTER.split(nullToEmpty(digest));
+        final Optional<String> hash = digestPairs.entrySet().stream()
+                    .filter(s -> s.getKey().toLowerCase().equals("sha1"))
+                    .map(Map.Entry::getValue)
+                    .findFirst();
+        return hash.map("urn:sha1:"::concat).orElse("");
```

Comment:
You can even go further and simply `.map(...).orElse("")` off of the `.findFirst()` line:

    .findFirst()
    .map("urn:sha1:"::concat)
    .orElse("");

Then you don't even need to define the `hash` value or import `Optional`.